exchanges/asset/asset.go

```diff
@@ -79,35 +80,24 @@ func (a Items) JoinToString(separator string) string {
 
 // IsValid returns whether or not the supplied asset type is valid or
 // not
-func IsValid(input Item) bool {
-	a := Supported()
-	for x := range a {
-		if strings.EqualFold(a[x].String(), input.String()) {
+func (a Item) IsValid() bool {
+	for x := range supported {
+		if supported[x].String() == a.String() {
```

Comment:
While the `New` function definitely helps with consistency, it would still be useful to test for valid equality without case, either by `ToLower()` or `strings.EqualFold(supported[x].String(), a.String())`. I can still do something silly like `asset.Item("SPOT")` and this will fail. See this:
```
func TestIsValid(t *testing.T) {
	if Item("rawr").IsValid() {
		t.Fatal("TestIsValid returned an unexpected result")
	}
	if !Item("SPOT").IsValid() {
		t.Fatal("TestIsValid returned an unexpected result")
	}
	if !Spot.IsValid() {
		t.Fatal("TestIsValid returned an unexpected result")
	}
}
```

Reply:
Yeah, I thought that but I wanted to specifically match to consts so we always have consistency when utilizing hash lookups. **SPOT is not a valid case in my update.** 
In a real-world context, we would directly access the exported package const list asset.Spot etc but for things like grpc or user level insertion we would specifically call New. 

I can change this though if it is specifically needed for an edge case I haven't considered.

Reply:
I like that. It prevents sneaky custom asset types and things like `SpOt` from getting into the database